n2survey/lime/survey.py

```diff
@@ -280,6 +306,115 @@ def get_responses(
+        """Retrieve or slice the responses DataFrame
+
+        Args:
+            key (pd.Series, pd.DataFrame, str, or tuple): A key for filter or get_responses method
+
+        Returns:
+            LimeSurvey or pd.DataFrame: A copy of LimeSurvey instance with filtered responses
+                DataFrameor or sliced responses DataFrame
+        """
+        # A bool-valued Series e.g. survey.responses["A3"] == "A5"
+        if isinstance(key, (pd.Series, pd.DataFrame)):
+            return self.filter(key)
+        # A question id e.g. "A3"
+        elif isinstance(key, str):
+            return self.get_responses(key)
+        # Two args e.g. survey.responses["A3"] == "A5", "B1"
+        # or 1:10, "B1"
+        elif isinstance(key, tuple):
+            rows, columns = key
+            return self.filter(rows).get_responses(columns)
+        else:
+            raise SyntaxError(
+                "Argument must be of type pd.Series, pd.DataFrame, str, or tuple."
+            )
+
+    def keep_choices(
+        self, question: str, choices: Union[str, bool, list], logic: str = "or"
+    ) -> "LimeSurvey":
+        """Filter responses DataFrame by keeping certain choices to specified questions
+
+        Args:
+            conditions (tuple or list of tuples): Conditions to re-format. Tuples in
+                format of (question_id, choices). choices can be a single str or list
+                of str. Multiple conditions are input as a list of tuples.
+                E.g. ("A6", "Woman") or [("A6", ["Woman", "Man"]), ("B6_SQ001", "Yes")]
+
+        Returns:
+            LimeSurvey: LimeSurvey with filtered responses
+        """
+
+        logic_dict = {"and": " & ", "or": " | "}
+        question_type = self.get_question_type(question)
+
+        # Convert into list
+        if not isinstance(choices, list):
+            choices = [choices]
+
+        # Handle multiple-choice questions with choices as sub-question
+        # labels differently, e.g. keep_choices("A3", "Chemistry")
+        if (question_type == "multiple-choice") & (True not in choices):
+            inverted_answer_dict = {
+                answer: id for id, answer in self.get_choices(question).items()
+            }
+            subquestions = [inverted_answer_dict[choice] for choice in choices]
+            return self.query(
+                logic_dict[logic].join(
+                    [f"{subquestion} == 'Y'" for subquestion in subquestions]
+                )
+            )
+        # All other cases, including single-choice questions, array sub-questions,
+        # mutliple-choice sub-questions with True labels
+        else:
+            return self.filter(
+                self.get_responses(question, labels=False)
+                .isin(choices)
+                .any(axis="columns")
+                | self.get_responses(question, labels=True)
+                .isin(choices)
+                .any(axis="columns")
+            )
```

Comment:
I don't think the logic is relevant here because you have previously said that filtering is performed on only a single question each time, and multiple-question filtering should be done by chaining `.filter()` or `.query()`. Therefore I don't think it makes sense to allow for `.all()` across multiple columns here.